src/re_unicode.rs

```diff
@@ -1034,6 +1034,40 @@ pub trait Replacer {
     fn no_expansion<'r>(&'r mut self) -> Option<Cow<'r, str>> {
         None
     }
+
+    /// Return a `Replacer` that borrows and wraps this `Replacer`.
+    ///
+    /// This is useful when you want to take a generic `Replacer` (which might
+    /// not be cloneable) and use it without consuming it, so it can be used
+    /// more than once:
+    ///
+    /// ```
```

Comment:
Sorry to nit at these things, but could you add an `# Example` header here to be consistent with how other examples are done? (I believe this is also the style in `std` too.)